lib/confg.rb

```diff
@@ -17,43 +27,49 @@ def erb_function(function_name, &block)
       self
     end
 
-    def configure(raise_on_missed_key = false)
-      @configuration ||= ::Confg::Configuration.new(raise_on_missed_key)
+    def configure
+      @configuration ||= ::Confg::Configuration.new(env: env, root: root)
       yield @configuration if block_given?
       @configuration
     end
-    alias_method :config, :configure
+    alias config configure
 
     def method_missing(method_name, *args, &block)
       config.send(method_name, *args, &block)
     end
 
-    def respond_to_missing?(*args)
+    def respond_to_missing?(*_args)
       true
     end
 
-    def get(path)
-      thing = self
-      path.split('.').each do |piece|
-        thing = thing.try(piece)
-      end
-      thing
-    end
-
     protected
 
     def calc_root_string
       return Rails.root.to_s if defined?(Rails)
       return RAILS_ROOT      if defined?(RAILS_ROOT)
       return RACK_ROOT       if defined?(RACK_ROOT)
 
-      ENV['RAILS_ROOT'] || ENV['RACK_ROOT'] || Dir.pwd
+      ENV["RAILS_ROOT"] || ENV["RACK_ROOT"] || Dir.pwd
+    end
+
+    def calc_root_path
+      ::Pathname.new(calc_root_string).expand_path
+    end
+
+    def calc_env_string
+      return ::Rails.env.to_s if defined?(::Rails)
+      return RAILS_ENV if defined?(RAILS_ENV)
+      return RACK_ENV if defined?(RACK_ENV)
+
+      ENV["RAILS_ENV"] || ENV["RACK_ENV"] || nil
```

Comment:
how would this work with engines? the engine would call the path relative to the rails root?